### DISCUSSION TITLE
Change of dirtyFields causes Material-U's TextField to lose focus

### DISCUSSION THREAD

**Discussion by harisbotic** [react-hook-form/react-hook-form#4173] [Q&A]
**Describe the bug**
When I make a change to an input in the form, I lose focus from the element and then I need to re-focus him manually again. This only happens when I listen to formState.dirtyFields state, when I don't listen to it, everything works fine. To clarify, once I re-focus the element after I lost it, I don't have this issue for that particular element anymore because it's already been added to dirtyFields thus it doesn't cause a re-render. (my guess)

**To Reproduce**
1. Start changing a field after a fresh start 

**Codesandbox link**
https://codesandbox.io/s/react-hook-form-controller-template-forked-jtyie?file=/src/index.js

**Desktop (please complete the following information):**
- OS: [mac]
- Browser [chrome]

**Comment by bluebill1049** [comment]
What's the issue here? and how to reproduce? It's not very clear.

**Comment by harisbotic** [comment]
sorry about that, I though it's clear enough.

Start typing in one of the TextFields and you will lose focus on that element. You will have to click on the element again to continue typing.

If you don't see the issue, I will record a loom video.

Thanks for fast response :D

**Comment by bluebill1049** [comment]
oh i see, move all your inside components outside the main, it's getting unmount and remounted each render.

**Comment by harisbotic** [comment]
@bluebill1049 I updated the codesandbox, but sadly I still have the same issue.
Could you please clarify more what you meant with "outside the main".

I created a new component called "FormContent" and put all tabs and input fields instead that.
The "main" component now only renders form, submit button and new "FormContent" component

**Comment by harisbotic** [comment]
after some more testing, I noticed that once we remove tabs, everything is working fine

**Comment by harisbotic** [comment]
@bluebill1049 any ideas of a workaround?

**Comment by bluebill1049** [comment]
like I suggested above, because your components is getting mount and unmounted:

```
/* eslint-disable react/prop-types, react/jsx-props-no-spreading */
  const TabPanel = (props) => {
    const { children, value, index, ...other } = props;

    return (
      <div
        role="tabpanel"
        hidden={value !== index}
        id={`vertical-tabpanel-${index}`}
        aria-labelledby={`vertical-tab-${index}`}
        {...other}
      >
        {value === index && (
          <Box pl={3}>
            <div>{children}</div>
          </Box>
        )}
      </div>
    );
  };
  /* eslint-enable */

  const tabLabels = tabs.map((tab) => {
    return (
      <Tab
        className={classes.editionTab}
        key={tab.label}
        label={<div className={classes.editTabLabel}>{tab.label}</div>}
      />
    );
  });

  const tabContent = tabs.map((tab, position) => {
    return (
      <TabPanel
        key={tab.label}
        value={tabIndex}
        index={position}
        className={classes.contentWrapper}
      >
        {tab.component}
      </TabPanel>
    );
  });
  
```

move those into their own components.

**Comment by harisbotic** [comment]
@bluebill1049 but that is exactly what I did, the component is called "FormContent" and the glitch is still there.

please take a look at new return method of the main component


```
  return (
    <div className={classes.containerWrapper}>
      <Container>
        <FormProvider {...formMethods}>
          <form onSubmit={handleSubmit(onSubmit)} encType="multipart/form-data">
            <Paper className={classes.paper}>
              <Paper className={classes.applyChangesTab}>
                <Button className={classes.applyChanges} type="submit">
                  Apply Changes
                </Button>
              </Paper>

              <FormContent />        // <-------------------------Tabs and inside tabs TextFields
            </Paper>
          </form>
        </FormProvider>
      </Container>
    </div>
  );
} 
```

I updated the sandbox

**Comment by bluebill1049** [comment]
still the same codesandbox...

**Comment by harisbotic** [comment]
cached prob or something...

forked link: https://codesandbox.io/s/react-hook-form-controller-template-forked-gxbbo

**Comment by bluebill1049** [comment]
<img width="805" alt="Screen Shot 2021-02-12 at 9 01 51 am" src="https://user-images.githubusercontent.com/10513364/107704179-f3ac6d00-6d10-11eb-9fe7-fa4e20341e22.png">
still there.
